include/Core/Transform/Rect.hpp

```diff
@@ -155,7 +155,12 @@ namespace obe::Transform
          * \asproperty
          */
         [[nodiscard]] double height() const;
-
+        /**
+        * \brief Checks if the Rect intersects a specified Rect
+        * \param other The other Rect to check
+        * \return true if the Rect intersects the other Rect, false otherwise
+        */
+        [[nodiscard]] bool doesIntersects(const Rect& rect) const;
```

Comment:
Duplicate of intersects method ?

Reply:
It return a boolean if there is an intersection, but doesn't calculate the Rect intersection.

Reply:
I'm not sure the overhead of `intersects` is worth doing another method where you can actually do :
```cpp
if (rect1.intersects(rect2))
{
   ...
}
```
Thanks to `std::optional` bool cast

Reply:
I made this function to avoid doing the Rect calculation of the intersection, because the AABB filter doesn't need it. I tried using doesIntersects() and intersects() on AABB filtering on a same benchmark scenario, and with doesIntersects() we get ~0.30% less Total CPU usage on getMaximumDistanceBeforeCollision().